### ISSUE TITLE
ConfiguredPermissions and manual request methods 

### ISSUE THREAD

**Issue by nmccann** [nickoneill/PermissionScope#130]
Hi,

I'm trying to integrate PermissionScope with an existing UI for which the documentation suggests using the various `request...` methods. When doing this I initially ran into a problem when requesting notifications - my `onAuthChange` callback wasn't being called. After some digging it appears that this occurred because the `configuredPermissions` array used in `getResultsForConfig` was empty. 

I was able to resolve this by calling `addPermission` but that seems odd for two reasons:
- Adding a permission without a message causes a failed assertion, but including a message isn't useful as my existing UI already has a message.
- It would be possible to infer what permissions I'm asking for based on the `request...` method being executed, so it seems like duplication.

At the very least it would be good to make the documentation more explicit about the need for `addPermission` when using `request...` - I struggled with this for a while.

That being said, the framework is very nice - thank you for the work you've put into it.

**Comment by nickoneill** [comment]
Hey there @nmccann, thanks for the feedback.

The manual request methods are definitely the way to go here but I didn't even think people would use them in conjunction with the `onAuthChange` callback so behavior there is currently undefined. One should not have to `addPermission` before using the manual methods! We'll for sure fix this in an upcoming release. Related to #125.

**Comment by nmccann** [comment]
Interesting - the "calling request\* methods directly" section of the documentation indicates that `onAuthChange` should be set.

**Comment by nickoneill** [comment]
Yep, the perils of working on software with a team! Some things are just assumed to happen :) We'll fix it up.

**Comment by nmccann** [comment]
lol well thank you for the quick response :)

**Comment by bre7** [comment]
`request`/`status` methods should include the necessary parameters then: e.g. notificationCategories for Notifications, etc.
